updater/src/updater.rs

```diff
@@ -338,9 +340,13 @@ impl fetch::urlhint::ContractClient for Updater {
 			.ok_or_else(|| "Registrar not available".into())
 	}
 
-	fn call(&self, address: Address, data: Bytes) -> Result<Bytes, String> {
-		self.client.upgrade().ok_or_else(|| "Client not available".to_owned())?
-			.call_contract(BlockId::Latest, address, data)
+	fn call(&self, address: Address, data: Bytes) -> BoxFuture<Bytes, String> {
+		let res = (move || {
+			self.client.upgrade().ok_or_else(|| "Client not available".to_owned())?
```

Comment:
I used the closure because of the `?` operator usage here, it seemed easier than doing an `and_then`